File: scripts/check-custom-claims.py
import json
from pathlib import Path
from firebase_admin import auth, credentials, initialize_app
import sys

if len(sys.argv) != 2:
    print("Usage: python scripts/check-custom-claims.py <uid>")
    sys.exit(1)

uid = sys.argv[1]
key_path = Path("c:/Users/onish/Downloads/finance-platform-362a5-c7cfbacbfa0a.json")
if not key_path.exists():
    print(f"Service account key not found: {key_path}")
    sys.exit(1)

service_account = json.loads(key_path.read_text(encoding="utf-8"))
cred = credentials.Certificate(service_account)
initialize_app(cred)

try:
    user = auth.get_user(uid)
    print("Custom claims:", json.dumps(user.custom_claims or {}, indent=2, ensure_ascii=False))
except Exception as err:
    print("Failed to fetch user:", err)
    sys.exit(1)

File: scripts/check-custom-claims.js
import admin from "firebase-admin";
import fs from "fs";

const [, , uid] = process.argv;
if (!uid) {
  console.error("Usage: node scripts/check-custom-claims.js <uid>");
  process.exit(1);
}

const keyPath = "c:/Users/onish/Downloads/finance-platform-362a5-c7cfbacbfa0a.json";
if (!fs.existsSync(keyPath)) {
  console.error(`Service account key not found at ${keyPath}`);
  process.exit(1);
}

const serviceAccount = JSON.parse(fs.readFileSync(keyPath, "utf-8"));

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  projectId: serviceAccount.project_id,
});

admin
  .auth()
  .getUser(uid)
  .then((user) => {
    console.log("Custom claims:", user.customClaims || {});
    process.exit(0);
  })
  .catch((error) => {
    console.error("Failed to fetch user:", error);
    process.exit(1);
  });
